session/recursive.go

```diff
@@ -0,0 +1,105 @@
+package session
+
+import (
+	"sort"
+	"strings"
+
+	"launchpad.net/gozk"
+)
+
+var defaultACL = zookeeper.ACL{
+	Perms:  zookeeper.PERM_ALL,
+	Scheme: "world",
+	Id:     "anyone",
+}
+
+// ChildrenRecursive returns a slice all of a node's descendants that are at
+// most `maxDepth` levels away from the root.
+func (s *ZKSession) ChildrenRecursive(path string, maxDepth int) ([]string, error) {
+	rootDepth := 0
+	if path != "/" {
+		rootDepth = strings.Count(path, "/")
+	}
+
+	stat, err := s.Exists(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if stat == nil {
+		return []string{}, nil
+	}
+
+	nodes := []string{path}
+	for index := 0; index < len(nodes); index++ {
```

Comment:
This won't work because I'm growing the `nodes` slice with new children nodes in the loop. It's a little messy, but the only alternatives that come to mind would be just as messy. Feel free to let me know if there's a nicer way to do this in go.